docs: Serve-docs target for locally previewing docs is broken

This commit https://github.com/argoproj/argo-rollouts/commit/96ba0302f4d904e9aba36c3341025f4f51eaed81 removed the plugin-docs Makefile target.

But that target is still mentioned as a dependency for `serve-docs`

So now the command make serve-docs stops with an error

```
argo-rollouts git:(master) make serve-docs
make: *** No rule to make target 'plugin-docs', needed by 'serve-docs'.  Stop.
```

---
<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->
**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

--- a/Makefile
+++ b/Makefile
@@ -268,7 +268,7 @@ release-docs: docs
 
 # convenience target to run `mkdocs serve` using a docker container
 .PHONY: serve-docs
-serve-docs: plugin-docs
+serve-docs: docs
 	docker run --rm -it -p 8000:8000 -v ${CURRENT_DIR}:/docs squidfunk/mkdocs-material serve -a 0.0.0.0:8000
 
 .PHONY: release-precheck

